docs/user/modeling.md

```diff
@@ -0,0 +1,144 @@
+
+!!! note
+    If you want to sync data into a model provided by a [Nautobot App](https://docs.nautobot.com/projects/core/en/stable/plugins/), you need to navigate to its respective source code repository. Furthermore, you may need to manually import the model if you're using the Nautobot Shell approach.
+
+## Custom Fields
+
+For [custom fields](https://docs.nautobot.com/projects/core/en/stable/models/extras/customfield/), you will need to use the `nautobot_ssot.contrib.CustomFieldAnnotation` class. Given a custom field called "Test" on the circuit provider model, this is how you could map that custom field into the diffsync model field `test_custom_field`:
+
+```python
+from typing_extensions import Annotated  # Python<3.9 - you need to install this module
+# from typing import Annotated  # Python>=3.9
+from nautobot.circuits.models import Provider
+from nautobot_ssot.contrib import NautobotModel, CustomFieldAnnotation
+
+class ProviderModel(NautobotModel):
+    _model = Provider
+    _modelname = "provider"
+    _identifiers = ("name",)
+    _attributes = ("custom_field",)
+
+    name: str
+    test_custom_field: Annotated[str, CustomFieldAnnotation(name="Test")]
+```
+
+!!! note
+    Defining a `CustomFieldAnnotation` variable is necessary since custom field names may include spaces, which are un-representable in Python object field names.
+
+## Many-to-one Relationships
+
+For many-to-one relationships (i.e. [foreign keys](https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_one/) or [generic foreign keys](https://docs.djangoproject.com/en/3.2/ref/contrib/contenttypes/#generic-relations)) a slightly different approach is employed. We need to add a field on our model for each field that we need in order to uniquely identify our related object behind the many-to-one relationship. We can do this using a [familiar synta](https://docs.djangoproject.com/en/3.2/topics/db/queries/#lookups-that-span-relationships) of double underscore separated paths. Assuming we want to synchronize prefixes and associate them with locations, we may be faced with the problems that locations aren't uniquely identified by name alone, but rather need the location type as well, we can address this as follows.
+
+```python
+from nautobot.ipam.models import Prefix
+from nautobot_ssot.contrib import NautobotModel
+
+class PrefixModel(NautobotModel):
+    _model = Prefix
+    _identifiers = ("network", "prefix_length")
+    _attributes = ("location__name", "location__location_type__name")
```

Comment:
so, the magic is that the model inspection will understand that `location` is a FK, and with these attributes we help it to identify the target, isn't it?

Reply:
correct!

Reply:
I have added an example query to further illustrate this